src/Compilers/Core/Portable/InternalUtilities/HashAlgorithms.cs

```diff
@@ -17,39 +18,60 @@ internal abstract class HashAlgorithm : IDisposable
 
         private readonly IDisposable _hashInstance;
 
+        private const string MscorlibAssembly = "mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089";
+        private const string HashingAssembly = "System.Security.Cryptography.Hashing, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a";
+        private const string HashingAlgorithmsAssembly = "System.Security.Cryptography.Hashing.Algorithms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a";
+
         static HashAlgorithm()
         {
-            var type = Type.GetType("System.Security.Cryptography.HashAlgorithm, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089");
-            var methods = type.GetTypeInfo().GetDeclaredMethods("ComputeHash");
-
-            s_bytesMethod = (from m in methods
-                             let ps = m.GetParameters()
-                             where ps.Length == 1 && ps[0].ParameterType == typeof(byte[])
-                             select m).Single();
-
-            s_bytesOffsetCountMethod = (from m in methods
-                                        let ps = m.GetParameters()
-                                        where ps.Length == 3 && ps[0].ParameterType == typeof(byte[]) && ps[1].ParameterType == typeof(int) && ps[2].ParameterType == typeof(int)
-                                        select m).Single();
-
-            s_streamMethod = (from m in methods
-                              let ps = m.GetParameters()
-                              where ps.Length == 1 && ps[0].ParameterType == typeof(Stream)
-                              select m).Single();
+            Type type = GetType("System.Security.Cryptography.HashAlgorithm", new string[] { HashingAssembly, MscorlibAssembly });
+            Debug.Assert(type != null, "Could not find HashingAlgorithm");
+            if (type != null)
+            {
+                var methods = type.GetTypeInfo().GetDeclaredMethods("ComputeHash");
+
+                s_bytesMethod = (from m in methods
+                                 let ps = m.GetParameters()
+                                 where ps.Length == 1 && ps[0].ParameterType == typeof(byte[])
```

Comment:
Unrelated, but code could be a little simpler using GetRuntimeMethod extension instead of the linq queries: https://msdn.microsoft.com/en-us/library/system.reflection.runtimereflectionextensions.getruntimemethod(v=vs.110).aspx